### DISCUSSION TITLE
Add zod transform support

### DISCUSSION THREAD

**Discussion by yanmarinich** [mantinedev/mantine#1942] [Feature requests]
### What package has an issue

@mantine/form

### Describe the bug

I use Select component that accepts only string values but my schema with zodResolver uses number for the value. Select component doesn't have any mapping prop to prepare the value before validation and zod preprocess, transform functions have no effect. Validation is fine but when I call form.onSubmit after successful validation I get raw values that I had before validation even though the type was created directly from the schema which uses number but the actual value is string.

### In which browser did the problem occur

Chrome

### Do you know how to fix the issue

Yes

### Possible fix

It would be great if useForm hook paid attention to zod transform and other functions that change value

**Comment by neofight78** [comment]
It's worth noting this would include supporting the input and output types being different.

**Comment by pbio** [comment]
I believe the issue lies in ZodResolver: https://www.npmjs.com/package/mantine-form-zod-resolver?activeTab=readme

Did you find a workaround to this?
